pkg/utils/build/hacbs.go

```diff
@@ -0,0 +1,96 @@
+package build
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+func FetchTaskRunResult(pr *v1beta1.PipelineRun, pipelineTaskName string, result string) (string, error) {
+	for _, tr := range pr.Status.TaskRuns {
+		if tr.PipelineTaskName != pipelineTaskName {
+			continue
+		}
+		for _, trResult := range tr.Status.TaskRunResults {
+			if trResult.Name == result {
+				return strings.TrimSuffix(trResult.Value, "\n"), nil
+			}
+		}
+	}
+	return "", fmt.Errorf(
+		"result %q not found in TaskRuns of PipelineRun %s/%s", result, pr.ObjectMeta.Namespace, pr.ObjectMeta.Name)
+}
+
+func FetchImageTaskRunResult(pr *v1beta1.PipelineRun, pipelineTaskName string, result string) (string, error) {
+	for _, tr := range pr.Status.TaskRuns {
+		if tr.PipelineTaskName != pipelineTaskName {
+			continue
+		}
+		for _, trResult := range tr.Status.TaskRunResults {
+
+			if trResult.Name == "BASE_IMAGE_REPOSITORY" || trResult.Name == result {
+				return (trResult.Value), nil
+			}
+		}
+	}
+	return "", fmt.Errorf(
+		"result %q not found in TaskRuns of PipelineRun %s/%s", result, pr.ObjectMeta.Namespace, pr.ObjectMeta.Name)
+}
+
+func ValidateImageTaskRunResults(taskname string, result string) bool {
+	var re = regexp.MustCompile(`devfile/python`)
+	if taskname == "sanity-inspect-image" {
+		if !(re.MatchString(result)) {
+			Fail(fmt.Sprintf("Expected Result for Taskrun '%s', failed with '%s'", taskname, result))
+		}
+	}
+	return true
+}
+
+func ValidateTaskRunResults(taskname string, result string) bool {
+	var re = regexp.MustCompile(`devfile/python`)
+	if taskname == "sanity-inspect-image" {
```

Comment:
This part is probably not needed now that `ValidateImageTaskRunResults` has been added.

Reply:
done

Reply:
Thanks